extensions/security/deployment/src/test/java/io/quarkus/security/test/permissionsallowed/SecuredBean.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.security.test.permissionsallowed;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import io.quarkus.security.PermissionsAllowed;
+
+@ApplicationScoped
+public class SecuredBean {
+
+    @PermissionsAllowed(value = "ignored", permission = CustomPermissionWithStringArg.class, params = "record.propertyOne")
```

Comment:
If we want people to use permission annotations implemented by methods, we should not require that they provide `value` if actions are meaningless for method permissions. Perhaps we need to make `value` optional?

Reply:
> If we want people to use permission annotations implemented by methods, we should not require that they provide value if actions are meaningless for method permissions. Perhaps we need to make value optional?

I was thinking a same, but it's complex question. Making value optional would make this permission more general authorization check. `value` always contain permission name (in addition to actions) and it is passed to https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/Permission.html#%3Cinit%3E(java.lang.String) so the interface is determined by Java. However the constructor allows `nulls`.

I think this makes sense together with providing Quarkus `java.security.Permission` parent that wouldn't need to be added with `SecurityIdentityAugmentor`. So that user would have to define just (no identity augmentation):

```
public class CustomPermission extends QuarkusBetterNamedPermission {

  private final Param2 param;

  public CustomPermission(Param2 param) {
     this.param = param;
  }

  protected boolean isAuthorized() {
      // decide return based on the permission inner state
  }

}
```

and then

```
@PermissionsAllowed(permission=CustomPermission.class)
public Dto securedMethod(Param1 param1, Param2 param2) {

}
```

I think it can't be done in this PR as it would trigger long discussion on concrete implementation. I'll concentrate on addressing your other comments.

Reply:
@michalvavrik Nice idea, but indeed worth to be done in another PR, thanks